src/dynamics/video_player/player/player.js

```diff
@@ -1181,6 +1181,30 @@ Scoped.define("module:VideoPlayer.Dynamics.Player", [
                         }, this);
                         this.delegateEvents(null, this.floatHandler);
                         this.floatHandler.init();
+
+                        if (this.get("floating") && this.get("floatingoptions").mobile) {
+                            const floatingElement = this.floatHandler.element;
+                            const viewport = window.visualViewport;
+                            let lastScrollTop = 0;
+
+                            function viewportHandler() {
+                                let st = window.scrollY || document.documentElement.scrollTop;
+                                const offsetLeft = viewport.offsetLeft;
+                                let offsetTop = 0;
+                                if (st > lastScrollTop) {
+                                    offsetTop =
+                                        window.top.innerHeight -
+                                        viewport.height;
+                                } else {
+                                    offsetTop = viewport.offsetTop;
+                                }
+                                lastScrollTop = st <= 0 ? 0 : st;
+                                floatingElement.style.transform = `translate(${offsetLeft}px, ${offsetTop}px) scale(${
+                                    1 / viewport.scale
+                                })`;
+                            }
+                            window.visualViewport.addEventListener("scroll", viewportHandler);
+                        }
```

Comment:
Hi @3akat, we can make it a little bit simpler and it still works as expected:
```suggestion
                        if (this.get("floating") && this.get("floatingoptions").mobile) {
                            const floatingElement = this.floatHandler.element;
                            const viewport = window.visualViewport;

                            function viewportHandler() {
                                floatingElement.style.transform = `translate(${viewport.offsetLeft}px, ${viewport.offsetTop}px)`;
                            }
                            viewport.addEventListener("scroll", viewportHandler);
                        }
```

Reply:
Looks great! Thank you, fixed